src/main/java/org/apache/hadoop/security/UserGroupInformationShim.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.security;
+
+import javax.security.auth.Subject;
+import javax.security.auth.kerberos.KerberosPrincipal;
+
+import java.util.Set;
+
+import static org.apache.hadoop.security.UserGroupInformation.AuthenticationMethod.KERBEROS;
+
+public final class UserGroupInformationShim
+{
+    private UserGroupInformationShim() {}
+
+    public static Subject getSubject(UserGroupInformation userGroupInformation)
+    {
+        return userGroupInformation.getSubject();
+    }
+
+    public static UserGroupInformation createUserGroupInformationForSubject(Subject subject)
+    {
+        if (subject == null) {
```

Comment:
This project is for java 7. There is no such method. I can use Guava one though.